### PR TITLE
Mail verification

### PR DESCRIPTION
**Pull request by easwarmoorthy**

### CODE REVIEW

File: routes/api/user.js [hunk 1]
```diff
@@ -15,27 +17,58 @@ router.post('/register', async function(req, res, next) {
...
-  if(existingUser) {
-    res.status(401).send({message:"Email or Username should be unique"})
-  } else {
-    await userModel.create(user)
-    res.send({message: "User registered Successfully"})
+  const existingUser = await userModel.findOne({ $or: [{email:{$eq: req.body.email}}, {username:{$eq: req.body.username}}]})
+  if(!existingUser) {
+    const registerUser = await userModel.create(user)
+    const transporter = nodemailer.createTransport({
+      host: "smtp.gmail.com",
```
Comment (arku) [inline routes/api/user.js]: I have heard using Gmail SMTP is problematic.

May I know why you're planning to use Gmail SMTP server?

File: server.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ const express = require('express')
 const app = express()
 const morgan = require('morgan')
 const bodyParser = require('body-parser')
+const dotenv = require('dotenv').config()
```
Comment (arku) [inline server.js]: What's the point of holding the result of `config` in a variable?

File: routes/api/user.js [hunk 3]
```diff
@@ -15,27 +17,58 @@ router.post('/register', async function(req, res, next) {
     email: req.body.email,
```
Comment (arku) [inline routes/api/user.js:36]: How about trimming and lowercasing the user input?

File: routes/auth.js [hunk 4]
```diff
@@ -21,12 +21,15 @@ const isUser = async function(req, res, next) {
...
+  else {
+    res.send({message: "User not authorized, invalid token"})
+  }
 }
 
 const isAdmin = async function(req, res, next) {
   const token = req.header('Authorization')
   if(token) {
-    const user = await userModel.findOne({token: token},{admin: true})
+    const user = await userModel.findOne({ $and:[{token: token},{admin: true}]})
```
Comment (arku) [inline routes/auth.js]: Why not just use `await userModel.findOne({ token, admin: true })`?

Am I missing something? :thinking:

File: routes/api/event.js [hunk 5]
```diff
@@ -2,32 +2,41 @@ const express = require('express');
...
-  //var User = mongoose.model('User',userModel);
-  var query = userModel.find({},{username:1,_id:0});
-  query.exec(function(err,usernames){
-    if (err)
-      return next(err);
-    res.send(usernames);
-    });
-});*/
+router.get('/events', isUser, async function(req, res) {
+  const events = await eventModel.find({});
```
Comment (arku) [inline routes/api/event.js:20]: Do we want to send all events in the payload?

**Review by arku — CHANGES_REQUESTED** [review]
The code looks sane on a quick glance.

Please take a look at my comments when you get a chance.

File: routes/api/event.js (hunk 5 above)
Comment (easwarmoorthy) [reply routes/api/event.js:20]: For now, need to discuss with @jenisonleo, may be changed to pagination kind of functionality.

File: routes/auth.js (hunk 4 above)
Comment (easwarmoorthy) [reply routes/auth.js]: Will make the change

File: routes/api/user.js (hunk 3 above)
Comment (easwarmoorthy) [reply routes/api/user.js:36]: Validations are not implemented now. Need to implement for all routes with Joi later.

File: server.js (hunk 2 above)
Comment (easwarmoorthy) [reply server.js]: Will change this

File: routes/api/user.js (hunk 1 above)
Comment (easwarmoorthy) [reply routes/api/user.js]: For now, this works. please suggest some alternative which can be implemented later

Comment (easwarmoorthy) [reply routes/api/user.js]: Need to change this later.

**Pull request merged**